src/lib/parser/statement/splitter.rs

```diff
@@ -109,19 +120,40 @@ impl<'a> StatementSplitter {
             math_paren_level: 0,
         }
     }
+
+    fn get_statement(&mut self, character: u8) -> StatementVariant<'a> {
+        let variant = if self.flags.contains(Flags::AND) {
+                self.flags.remove(Flags::AND);
+                self.data = &self.data[self.read + 1..].trim();
+                StatementVariant::And(&self.data)
```

Comment:
This doesn't look like it's going to split statements correctly. It's going to return everything after `self.read + 1`, rather than up to.

Reply:
At that point `self.read` just points to `& echo true` in `matches Foo '([A-Z])\w+' && echo true`, so it makes sense just to returns everything after that which is `" echo true".trim()`

Reply:
Shells can have more elaborate statement-splitting requirements, such as

```sh
cmd arg; cmd arg && cmd this || cmd that; cmd arg
```